### PR TITLE
remove public constructor

### PR DESCRIPTION
**Pull request by dcousens**
straight forward... but I don't know if it is worth it.

https://github.com/cryptocoinjs/ecurve/blob/master/lib/names.js#L10-L16 would need to be updated.

And https://github.com/cryptocoinjs/ecurve/blob/master/lib/point.js#L4 changed to `BigInteger.fromNumber`

### CODE REVIEW

**Comment by dcousens** [comment]
:shrug: ... I don't even know what my motivation for this was

File: test/bigi.js [hunk 1]
```diff
@@ -6,72 +7,76 @@
...
 
-describe('BigInteger', function () {
-  it('should work without new', function() {
-    var bi = BigInteger('12345')
-    assert.equal(bi.toString(10), '12345')
-  })
+function newBI (str, radix) {
+  if (str[0] === '-') {
+    if (radix === 16) return BigInteger.fromString('' + parseInt(str, 16))
+  }
```
Comment (dcousens) [inline test/bigi.js]: We don't test any numbers < -MAX_SAFE_INT,  and `fromHex` only supports unsigned hex

**Comment by jprichardson** [comment]
I'm really okay with whatever here. I'm not really invested in bigi anymore and intend on moving most of everything to bn.js.

**Comment by dcousens** [comment]
@jprichardson same,  but it was an open issue in my list so I figured... I either close it or just do it.

**Comment by dcousens** [comment]
shall we bump to `2.0.0`?

**Pull request merged**